.github/workflows/clients-java-build.yml

```diff
@@ -46,13 +46,13 @@ jobs:
       run: mvn compile
 
     - name: Build
-      if: ${{ github.event_name == 'pull_request' }}
+      if: ${{ github.ref != 'refs/heads/main' }}
       working-directory: src/clients/java
       run: mvn -B package --file pom.xml
 
     - name: Save local package
       uses: actions/upload-artifact@v3
-      if: ${{ github.event_name == 'pull_request' }}
```

Comment:
Out of scope for this PR, but maybe we should have two different yamls:

* `ci.yml` which runs tests, gates bors, and is run on every branch (technically, with bors it isn't necessary to run tests on master, as that's redundant, but that's also a good way to catch flakes)
* `release.yml` which does the release, and is run on master branch only. 

Within bors paradigm, clearly separating "check that gate master branch" from "everything else we do automatically with known-good master branch" makes sense philosophically. 

Don't think there are any direct practical benefits though.

Reply:
I agree - bors aside, we currently gate releasing in different ways depending on the case (eg, in the node / go clients a whole job is gated behind `github.ref == 'refs/heads/main'` whereas the other clients do a different step within a job depending.)

The `build-and-push` step does it as a flag, which gave me a mini-fright the first time I skimmed over the config and didn't see it being conditional on `main` :smile: